lambdas/InspectMediaLambdaFunction/ebu-r-128.js

```diff
@@ -0,0 +1,85 @@
+const childProcess = require('child_process');
+const os = require('os');
+
+/**
+ * @typedef {object} FfmpegLoudnormResult
+ * @property {string} [input_i]
+ * @property {string} [input_tp]
+ * @property {string} [input_lra]
+ * @property {string} [input_thresh]
+ */
+
+module.exports = {
+  /**
+   * @param {string} filePath
+   * @returns {Promise<FfmpegLoudnormResult>}
+   */
+  inspect: async function inspect(filePath) {
+    return new Promise((resolve, reject) => {
+      const start = process.hrtime();
+
+      // This will return JSON data after a lot of other FFmpeg process
+      // information
+      const childProc = childProcess.spawn(
+        '/opt/bin/ffmpeg',
+        [
+          '-v',
+          'info',
+          '-hide_banner',
+          '-nostats',
+          '-i',
+          filePath,
+          '-af',
+          'loudnorm=dual_mono=true:print_format=json',
+          '-f',
+          'null',
+          '-',
+        ],
+        {
+          env: process.env,
+          cwd: os.tmpdir(),
+        },
+      );
+      const resultBuffers = [];
+
+      // The data we want seems to end up in stderr, so add that to the buffer
+      // capture as well
```

Comment:
Regardless of what I tried, the text output of the above FFmpeg command was triggering the `stderr.on` callback, not `stdout`. That was true of the `loudnorm` filter output, and also the regular file info that it puts out (i.e., the stuff that shows up when `-v info` is set).

This seems wrong, but as-is everything's working. Not sure how big a deal it is.

Reply:
I can't find anything either.

Reply:
well, I found this? https://stackoverflow.com/questions/58994439/isolate-the-output-only-ffmpeg